lib/Driver/Job.cpp

```diff
@@ -131,6 +191,20 @@ StringRef CommandOutput::getAdditionalOutputForType(types::ID Type) const {
   return getOutputForInputAndType(Inputs[0].Primary, Type);
 }
 
+SmallVector<StringRef, 16>
+CommandOutput::getAdditionalOutputsForType(types::ID Type) const {
+  SmallVector<StringRef, 16> V;
+  if (AdditionalOutputTypes.count(Type) != 0) {
+    for (auto const &I : Inputs) {
+      auto Out = getOutputForInputAndType(I.Primary, Type);
+      if (!Out.empty())
```

Comment:
I'm pretty sure you can assert this. We're already in the "additional outputs" section.

Reply:
yeah but we're looping _per input_. If an input doesn't have an additional output of a given type, getOutputForInputAndType will return an empty string. Those are the 0 and 1 cases.

Reply:
\*squint\* The only such case is WMO, and in that case the aux output paths are stored under the `""` entry in the OFM, not an individual file. Or am I missing something?

Reply:
Might be a bug elsewhere (and I realized now that this isn't the zero case -- the whole loop is guarded by AdditionalOutputTypes.count(Type) != 0) but there _is_ one case in the testsuite that exercises this. Driver/multi-threaded.swift produces a derived OFM with two .swift primary inputs and two .o primary outputs and only _one_ additional-output of type TY_Dependencies (main.d). It's possible we should be outputting two dependencies files there, but currently we're not. Maybe I should loop over inputs up in Driver.cpp in this case or something?

Reply:
No, that's a WMO test case. If there is a .d file being added, it should not be under any particular input.

(That said, what happens if an OFM has stray per-file entries in it when performing WMO? Do those end up in the derived file map?)